hadoop-plugin/src/main/groovy/com/linkedin/gradle/hdfs/HdfsFileSystem.groovy

```diff
@@ -192,6 +196,47 @@ class HdfsFileSystem {
   }
 
   /**
+   * The zipFile is the zip file on the local system. This file is extracted to the destDirectory.
+   * If destDirectory doesn't exist, then it is created.
+   *
+   * @param zipFile The zip file on local system
+   * @param destDirectory The destination directory on hdfs where zipFile should be extracted.
+   * @throws IOException If zipFile cannot be extracted to destDirectory
+   */
+  public void copyFromLocalZip(File zipFile, Path destDirectory) throws IOException {
+
+    // create the destination directory if it doesn't exists
+    if (!exists(destDirectory)) {
+      mkdir(destDirectory);
+    }
+
+    logger.info("zip file path: ${zipFile.toString()}");
+    logger.info("destination directory: ${destDirectory.toString()}");
+
+    ZipInputStream zipIn = new ZipInputStream(new FileInputStream(zipFile));
+    ZipEntry entry = zipIn.getNextEntry();
+
+    //iterates over entries in the zip file
+    while (entry != null) {
+      String filePath = destDirectory.toString() + File.separator + entry.getName();
```

Comment:
entry.getName gives us the name of the entry, if it is a file it will give the filename and similar for directory. We differentiate between directory and files in the zip by calling entry.isDirectory() method. When the entry is a directory, we create the directory in hdfs and when the entry is a file, we extract the file using extractFile method. This way the path information is preserved. I've tested this by uploading a zip containing various nested directories and the whole directory structure was extracted to the hdfs.